stdlib/http/src/main/ballerina/http/client_endpoint.bal

```diff
@@ -205,6 +207,8 @@ public type AuthConfig record {
     string tokenUrl;
     string clientId;
     string clientSecret;
+    CredentialBearer credentialBearer = AUTH_HEADER_BEARER;
+    string ^"scope";
```

Comment:
IMO, it's inconvenient and confusing for the user when they have to use the escaping sequence. Have we done that with any other config name? Since users will have to also escape that I'm +1 to use "requestedScope". 

I have moved this to mail discussion too.